common/util.c

```diff
@@ -64,3 +68,36 @@ int get_modifier_names(const char **names, uint32_t modifier_masks) {
 
 	return length;
 }
+
+pid_t get_parent_pid(pid_t child) {
+	pid_t parent;
+	char file_name[100];
+	char *buffer = NULL;
+	char *token = NULL;
+	const char sep[2] = " ";
+	FILE *stat = NULL;
+
+	sway_log(L_DEBUG, "trying to get parent pid for child pid %d", child);
+
+	sprintf(file_name, "/proc/%d/stat", child);
+
+	if (!(stat = fopen(file_name, "r")) || !(buffer = read_line(stat))) {
```

Comment:
Actually now that I look at this again I think `||` is right. Consider the case where `fopen()` succeeds but `read_line()` fails. In this case, with `&&` you'd have:

    if (not (fopen succeeds) and not (read_line does not succeed))
    if (not (true) and not (false))
    if (false and true)
    false
    don't return -1 to signal failure

With `||`:

    if (not (fopen succeeds) or not (read_line does not succeed))
    if (not (true) or not (false))
    if (false or true)
    true
    return -1 to signal failure